microcloud/cmd/microcloud/ask.go

```diff
@@ -756,7 +759,35 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 			}
 		}
 
-		uplink, ovn := lxd.DefaultOVNNetwork(ipv4Gateway, ipv4Ranges, ipv6Gateway)
+		var ipv4DNSNameServers string
+		var ipv6DNSNameServers string
+		for gateway, dnsAddr := range dnsConfig {
+			ip, _, err := net.ParseCIDR(gateway)
+			if err != nil {
+				return err
+			}
+
+			if ip.To4() != nil {
+				ipv4DNSNameServers = dnsAddr
+			} else {
+				ipv6DNSNameServers = dnsAddr
+			}
+		}
+
+		allDNSServers := ""
+		if ipv4DNSNameServers != "" {
+			allDNSServers = ipv4DNSNameServers
+		}
+
+		if ipv6DNSNameServers != "" {
+			if allDNSServers != "" {
+				allDNSServers += "," + ipv6DNSNameServers
+			} else {
+				allDNSServers = ipv6DNSNameServers
+			}
+		}
+
+		uplink, ovn := lxd.DefaultOVNNetwork(ipv4Gateway, ipv4Ranges, ipv6Gateway, allDNSServers)
```

Comment:
So here, it looks to me like you're fetching the results from each of the ipv4 and ipv6 dns questions, and then just appending them to the same list, which gets stored in the OVN config as `dns.nameservers`. Which says to me that you probably don't need 2 separate questions for the nameservers.

How about having just 1 question where the user can enter both ipv4 and ipv6 addresses? Then we won't need this extra IP parsing and version checking.

Reply:
@masnax @gabrielmougard yes agreed, we shouldn't ask the DNS question twice, just once and accept IPv4 and IPv6 addresses, this is what I mentioned a few days ago here:

https://github.com/canonical/microcloud/pull/228#discussion_r1403270125